test/test_autograd.py

```diff
@@ -1616,31 +1597,7 @@ def __del__(self):
                 gc.collect()
 
         for _ in range(10):
-            CollectOnDelete()(torch.randn(1, requires_grad=True)).backward()
-
-    def test_call_legacy_twice(self):
-        class Id(Function):
-            def forward(self, x):
-                self.save_for_backward(x)
-                return x
-
-            def backward(self, grad_x):
-                x = self.saved_tensors
-                return x
-
-        f = Id()
-        x1 = torch.zeros(1, requires_grad=True)
-        x2 = torch.ones(1, requires_grad=True)
-        y = f(x1)
-        with warnings.catch_warnings(record=True) as w:
-            z = f(x2)
-        self.assertIn('extending-torch-autograd', str(w[1].message))
-        # I don't really care about the functional correctness of this
-        # part of the test: if you make a change that causes this test
-        # to fail, it's probably OK to just fix this test case to follow
-        # it.  I'm mostly making sure we don't segfault here.
-        y.backward()
-        self.assertEqual(x2.grad, x2)
+            CollectOnDelete().forward(torch.randn(1, requires_grad=True)).backward()
```

Comment:
This will raise on error on construction no?

Reply:
no, it is not get called.

Reply:
What is this test doing if this is not called?

Reply:
No, I mean, __call__ is not triggered in CollectOnDelete().

Reply:
Is there any reason why we allow instantiating old-style Function?
You might want to error out in `FunctionMeta`'s `__init__` in function.py as well if you detect an old style Function.

Reply:
Nope, no spicial reason. I am just not sure whether we should disallow people creating Function obj. People can call forward using CollectOnDelete.forward() or CollectOnDelete.apply() directly, but there is nothing wrong for people doing CollectOnDelete().forward() as well. Please advise.